software/spmd/bsg_attr_noalias_remote_ex/main.cpp

```diff
@@ -0,0 +1,205 @@
+float B[N_ELS] __attribute__ ((section (".dram"))) = MACRO_B;
+float C[N_ELS] __attribute__ ((section (".dram"))) = {0};
+float gold[N_ELS] __attribute__ ((section (".dram")));
+
+float bsg_attr_remote _A[N_ELS] __attribute__ ((section (".dram"))) = MACRO_A;
+float bsg_attr_remote _B[N_ELS] __attribute__ ((section (".dram"))) = MACRO_B;
+float bsg_attr_remote _C[N_ELS] __attribute__ ((section (".dram"))) = {0};
+float bsg_attr_remote _gold[N_ELS] __attribute__ ((section (".dram"))) = {0};
+
+int main(){
+
+        bsg_set_tile_x_y();
+
+        if ((__bsg_x == 0) && (__bsg_y == 0)) {
+                saxpy(A, B, gold, MACRO_ALPHA); 
+                _saxpy(_A, _B, _gold, MACRO_ALPHA); 
+ 
+                // **************************************************
+                // Test the simple version of SAXPY
+                // **************************************************
+                bsg_printf("Testing SAXPY\n");
+                // Heat the cache (just for demonstration purposes)
+                saxpy(A, B, C, MACRO_ALPHA);
+                // Measure performance
+                bsg_cuda_print_stat_start(SAXPY_TAG_BASE);
+                saxpy(A, B, C, MACRO_ALPHA);
+                bsg_cuda_print_stat_end(SAXPY_TAG_BASE);
+                
+                if (memcmp(C, gold, N_ELS * sizeof(*C)))
+                        bsg_fail();
+                memset(C, 0, N_ELS * sizeof(*C));
+
+
+                // **************************************************
+                // Test the C version of SAXPY
+                // **************************************************
+                bsg_printf("Testing C SAXPY +unroll +noalias + remote\n");
+                // Heat the cache (just for demonstration purposes)
+                saxpy_c(_A, _B, _C, MACRO_ALPHA);
+                // Measure performance
+                bsg_cuda_print_stat_start(SAXPY_TAG_C);
+                saxpy_c(_A, _B, _C, MACRO_ALPHA);
+                bsg_cuda_print_stat_end(SAXPY_TAG_C);
+                
+#ifndef __llvm__
+                // LLVM complains about memory intrinsics...
+                if (memcmp(_C, _gold, N_ELS * sizeof(*C)))
+                        bsg_fail();
+                memset(_C, 0, N_ELS * sizeof(*C));
+#endif
+
+                // **************************************************
+                // Test the C version of SAXPY (with more unrolling)
+                // **************************************************
+                bsg_printf("Testing C SAXPY (MORE) +unroll +noalias + remote\n");
+                // Heat the cache (just for demonstration purposes)
+                saxpy_c_moreunroll(_A, _B, _C, MACRO_ALPHA);
+                // Measure performance
+                bsg_cuda_print_stat_start(SAXPY_TAG_C_MOREUNROLL);
+                saxpy_c(_A, _B, _C, MACRO_ALPHA);
+                bsg_cuda_print_stat_end(SAXPY_TAG_C_MOREUNROLL);
+                
+#ifndef __llvm__
```

Comment:
At the moment, no. Mostly because of this: 

```
main.cpp:179:17: error: no matching function for call to 'memset'
                memset(_C, 0, N_ELS * sizeof(*C));
                ^~~~~~
```